### PR TITLE
feat: add `enableFederation` flag to `server`

### PR DESCRIPTION
**Pull request by meskill**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #... _(Replace "..." with the issue number)_

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/core/config/transformer/subgraph.rs [hunk 1]
```diff
@@ -34,6 +34,13 @@ impl Transform for Subgraph {
     type Error = String;
 
     fn transform(&self, mut config: Self::Value) -> Valid<Self::Value, Self::Error> {
+        let enable_federation = config.server.enable_federation;
```
Comment (tusharmath) [inline src/core/config/transformer/subgraph.rs:37]:
```suggestion
        let enable_federation: bool = config.server.get_enable_federation(); 
```

Comment (tusharmath) [reply src/core/config/transformer/subgraph.rs:37]: Enable only if set to true. Disable by default.

**Comment by tusharmath** [comment]
- [ ] Create a PR for docs.
- [ ] Resolve conflicts.

File: src/core/config/transformer/subgraph.rs (hunk 1 above)
Comment (meskill) [reply src/core/config/transformer/subgraph.rs:37]: done

**Comment by meskill** [comment]
> * [x]  Create a PR for docs.
> * [x]  Resolve conflicts.

https://github.com/tailcallhq/tailcallhq.github.io/pull/524

**Review by tusharmath — APPROVED** [review]

**Pull request merged**